apm-agent-plugins/apm-logging-plugin/apm-logging-plugin-common/src/test/java/co/elastic/apm/agent/loginstr/reformatting/UtilsTest.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.loginstr.reformatting;
+
+import co.elastic.apm.agent.AbstractInstrumentationTest;
+import co.elastic.apm.agent.logging.LogEcsReformatting;
+import co.elastic.apm.agent.logging.LoggingConfiguration;
+import org.junit.jupiter.api.Test;
+
+import javax.annotation.Nullable;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+public class UtilsTest extends AbstractInstrumentationTest {
+
+    private static final String fileSeparator = System.getProperty("file.separator");
+
+    @Nullable
+    private final String logEcsFormattingDestinationDir = config.getConfig(LoggingConfiguration.class).getLogEcsFormattingDestinationDir();
+
+    private String computeReformattedLogFilePathWithConfiguredDir(String logFilePath) {
+        return Utils.computeLogReformattingFilePath(logFilePath, logEcsFormattingDestinationDir);
+    }
+
+    @Test
+    void testShadePathComputation() {
```

Comment:
[minor] seems a good candidate for changing the `computeReformattedLogFilePathWithConfiguredDir` entry point and test to use `Path` instead of a raw file path in a `String` that requires to rewrite path.